### ISSUE TITLE
Random choice for race is not persistent

### ISSUE THREAD

**Issue by ihhub** [ihhub/fheroes2#265]
Open the game and start a map where you don't choose any specific race for anyone. By default race is random. Start playing map.

<img width="752" alt="Screenshot 2019-12-24 at 9 20 22 AM" src="https://user-images.githubusercontent.com/19829520/71387395-afeb3580-262e-11ea-81da-a3e98994193f.png">

Create a new game again and the same map. You'd notice that all races are predetermined.

<img width="752" alt="Screenshot 2019-12-24 at 9 21 04 AM" src="https://user-images.githubusercontent.com/19829520/71387408-bd082480-262e-11ea-8bd8-6855f274cc9a.png">

We want to save random races until user choses some specific one.

**Comment by Killbrum** [comment]
I think that we can close the issue since it is not reproducible on the latest version. @ihhub

**Comment by Branikolog** [comment]
Issue is still on:
When we start a new game, and then return back to main menu, we can see, that all the races are set, as they were at previous game in current fheroes2.
In the original demo, for example, only player's race was saved in the same conditions. (Even if it was set random). Also in original demo, none of opponents' races was saved, even if we choose them manually.
The best implementation for this logic is like that:
If we start game with random AI race, the next time we start new game - it should be left as random. If we set AI race manually, it should be saved the next time we start new game. For example: we set 4 AI as knights and one random race AI. 
The next time we start new game we should also have the same preferences. 4 AI knights and one random race.
Considering human player, i think we should leave it as current. If we set manually human race, it should be saved and the next time we start new game, it should be the same as it was. 
If we set random human race, it also should be saved and the next time we start new game race should be defined by previous choice. (Ex: We set random race for human. Start new game. The game starts as a knight. We choose new game, and there's already set knight race for human.)
